resolwe/flow/tests/test_api.py

```diff
@@ -225,6 +243,58 @@ def test_add_remove_data(self):
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         self.assertEqual(c.data.count(), 0)
 
+    def test_delete(self):
+        collection = Collection.objects.create(
+            name="Test collection",
+            contributor=self.contributor,
+        )
+
+        data_1, data_2 = self._create_data(), self._create_data()
+        entity_1, entity_2 = self._create_entity(), self._create_entity()
+
+        collection.data.add(data_1, data_2)
+        collection.entity_set.add(entity_1, entity_2)
+
+        assign_perm("view_collection", self.user, collection)
+        assign_perm("edit_collection", self.user, collection)
+        assign_perm("view_data", self.user, data_1)
+        assign_perm("view_data", self.user, data_2)
+        assign_perm("edit_data", self.user, data_1)
+        assign_perm("view_entity", self.user, entity_1)
+        assign_perm("view_entity", self.user, entity_2)
+        assign_perm("edit_entity", self.user, entity_1)
+
+        request = factory.delete(self.detail_url(collection.pk))
+        force_authenticate(request, self.user)
+        self.collection_detail_viewset(request, pk=collection.pk)
+
+        self.assertTrue(Data.objects.filter(pk=data_1.pk).exists())
+        self.assertTrue(Data.objects.filter(pk=data_2.pk).exists())
+        self.assertTrue(Entity.objects.filter(pk=entity_1.pk).exists())
+        self.assertTrue(Entity.objects.filter(pk=entity_2.pk).exists())
+
+        # Recreate the initial state and test with `delete_content` flag.
+        collection = Collection.objects.create(
+            name="Test collection",
+            contributor=self.contributor,
+        )
+
+        collection.data.add(data_1, data_2)
+        collection.entity_set.add(entity_1, entity_2)
+
+        assign_perm("view_collection", self.user, collection)
+        assign_perm("edit_collection", self.user, collection)
+
+        request = factory.delete('{}?delete_content=1'.format(self.detail_url(collection.pk)))
+        force_authenticate(request, self.user)
+        self.collection_detail_viewset(request, pk=collection.pk)
+
+        # Only objects with `edit` permission can be deleted.
```

Comment:
But in this case, some objects are deleted while others are silently kept and (for the caller) there is no way to know that this happened? This is also inconsistent with how filesystem permissions behave.